spec/helpers/http_mock/asserts.lua

```diff
@@ -0,0 +1,153 @@
+local setmetatable = setmetatable
+local ipairs = ipairs
+local pairs = pairs
+local pcall = pcall
+local error = error
+
+local http_mock = {}
+
+local build_in_checks = {}
+
+local eventually_MT = {}
+eventually_MT.__index = eventually_MT
+
+local step_time = 0.01
+
+-- example for a check function
+-- local function(session, status)
+--   -- must throw error if the assertion is not true
+--   -- instead of return false
+--   assert.same(session.resp.status, status)
+--   -- return a string to tell what condition is satisfied
+--   -- so we can construct an error message for reverse assertion
+--   -- in this case it would be "we don't expect that: has a response with status 200"
+--   return "has a response with status " .. status
+-- end
+
+local function eventually_has(check, mock, ...)
+  local time = 0
+  local ok, err
+  while time < mock.eventually_timeout do
+    local records = mock:get_record()
+    for _, record in ipairs(records) do
+    -- use pcall so the user may use lua assert like assert.same
+      ok, err = pcall(check, record, ...)
+      if ok then
+        return true
+      end
+    end
+    time = time + step_time
```

Comment:
It does not seem as if `step_time` is actually a time in seconds and that `eventually_timeout` is more like a "maximum repeat" counter.  Consider switching to integers and renaming `time` and `timeouts` to reflect that there is a retry counter.  Is it good to spin waiting for the result like this?  Would a little delay and a properly implemented timeout not be better?

Reply:
You're right. Forgot to add the delay logic.

Reply:
It would be more precise to get the current time as `ngx.sleep` could sleep longer than requested.  Not a blocker though.